app/controllers/api/v2/registration_controller.rb

```diff
@@ -0,0 +1,104 @@
+module Api
+  module V2
+    class RegistrationController < V2::BaseController
+      include Api::Version2
+      include Foreman::Controller::Registration
+      include Foreman::Controller::Parameters::Registration
+
+      rescue_from ActiveRecord::RecordNotFound do |error|
+        logger.info "#{error.message} (#{error.class})"
+        not_found(error.message)
+      end
+
+      rescue_from ActiveRecord::RecordInvalid do |error|
+        render_error(error.message, status: :unprocessable_entity)
+      end
+
+      api :GET, '/register', N_('Render Global registration template')
+      param :organization_id, :number, desc: N_("ID of the Organization to register the host in.")
+      param :location_id, :number, desc: N_("ID of the Location to register the host in.")
+      param :hostgroup_id, :number, desc: N_("ID of the Host group to register the host in.")
+      param :operatingsystem_id, :number, desc: N_("ID of the Operating System to register the host in.")
+      def global
+        find_global_registration
+
+        unless @provisioning_template
+          not_found _('Global Registration Template with name %s defined via default_global_registration_item Setting not found, please configure the existing template name first') % Setting[:default_global_registration_item]
+          return
+        end
+
+        render plain: @provisioning_template.render(variables: @global_registration_vars).html_safe
+      end
+
+      api :POST, "/register", N_("Find or create a host and render the Host registration template")
+      param :name, String, required: true
+      param :location_id, :number, required: true
+      param :organization_id, :number, required: true
+      param :ip, String, desc: N_("not required if using a subnet with DHCP proxy")
```

Comment:
Thinking out loud here: can we do better than just string? I don't know rails well enough if we can just use `IPAddr` and actually validate the IP syntactically as well. It's likely I'm missing something, but some education here would be nice. For example, where can I read docs on this?

Reply:
This can work out of the box since we don't enfroce parameters validation. However it may be a good idea to convert the storage to native IP address. If that does not cause problems on sqlite in jenkins.